packages/regl-worldview/src/commands/GLTFScene.js

```diff
@@ -0,0 +1,210 @@
+    if (drawCalls) {
+      return;
+    }
+
+    const textures = model.json.textures.map((texture) => {
+      const sampler = model.json.samplers[texture.sampler];
+      return regl.texture({
+        data: model.images[texture.source],
+        min: "linear", //glConstantToRegl(sampler.minFilter),
+        mag: glConstantToRegl(sampler.magFilter),
+        wrapS: glConstantToRegl(sampler.wrapS),
+        wrapT: glConstantToRegl(sampler.wrapT),
+      });
+    });
+    drawCalls = [];
+
+    function drawMesh(mesh, nodeMatrix) {
+      for (const primitive of mesh.primitives) {
+        const material = model.json.materials[primitive.material];
+        const texInfo = material.pbrMetallicRoughness.baseColorTexture;
+        drawCalls.push({
+          indices: model.accessors[primitive.indices],
+          positions: model.accessors[primitive.attributes.POSITION],
+          normals: model.accessors[primitive.attributes.NORMAL],
+          texCoords: model.accessors[primitive.attributes[`TEXCOORD_${texInfo.texCoord || 0}`]],
+          baseColorTexture: textures[texInfo.index],
+          nodeMatrix,
+        });
+      }
+    }
+    function drawNode(node, parentMatrix) {
+      const nodeMatrix = node.matrix
+        ? mat4.clone(node.matrix)
+        : mat4.fromRotationTranslationScale(
+            mat4.create(),
+            node.rotation || [0, 0, 0, 1],
+            node.translation || [0, 0, 0],
+            node.scale || [1, 1, 1]
+          );
+      mat4.mul(nodeMatrix, parentMatrix, nodeMatrix);
+      if (node.mesh != null) {
+        drawMesh(model.json.meshes[node.mesh], nodeMatrix);
+      }
+      if (node.children) {
+        for (const childIdx of node.children) {
+          drawNode(model.json.nodes[childIdx], nodeMatrix);
+        }
+      }
+    }
+    for (const nodeIdx of model.json.scenes[model.json.scene].nodes) {
+      const rootTransform = mat4.create();
+      mat4.rotateX(rootTransform, rootTransform, Math.PI / 2);
+      mat4.rotateY(rootTransform, rootTransform, Math.PI / 2);
+      drawNode(model.json.nodes[nodeIdx], rootTransform);
+    }
+  }
+
+  const withPoseMatrix = regl({
+    context: {
+      poseMatrix: (context, props) =>
+        mat4.fromRotationTranslationScale(
+          mat4.create(),
+          orientationToVec4(props.pose.orientation),
+          pointToVec3(props.pose.position),
+          props.scale ? pointToVec3(props.scale) : [1, 1, 1]
+        ),
+    },
+  });
+
+  return (props) => {
+    prepareDrawCallsIfNeeded(props.model);
+    withPoseMatrix(props, () => {
+      command(drawCalls);
+    });
+  };
+};
+
+type Props = {|
+  modelURL: string,
+  children: {|
+    pose: Pose,
+    scale: Point,
+  |},
+|};
+
+export default class GLTFScene extends React.Component<Props, *> {
+  state = {
+    model: undefined,
+  };
+  _context = undefined;
+
+  componentDidMount() {
+    loadGLB(this.props.modelURL).then((model) => {
+      this.setState({ model });
+      if (this._context) {
```

Comment:
This is an unfortunate oversight in the API that we need to call `this._context.onDirty` when the child `<Command >` props change.  This is totally fine for now, but just fyi I have some unfinished work to make this unneeded in the future.  IMO if `<Command />` receives new props it should trigger a draw call.  To prevent over-rendering the parent should implement `shouldComponentUpdate` if needed.  I think this is more in line w/ how one would expect react to work: render unless told specifically not to.